test/nvexec/transfer_when_all.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2023 NVIDIA Corporation
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <test_common/type_helpers.hpp>
+#include <test_common/receivers.hpp>
+
+#include <exec/inline_scheduler.hpp>
+#include <nvexec/stream_context.cuh>
+#include <stdexec/execution.hpp>
+
+#include <catch2/catch.hpp>
+
+namespace ex = stdexec;
+
+TEST_CASE("transfer_when_all returns a sender", "[cuda][stream][adaptors][transfer_when_all]") {
+  nvexec::stream_context stream_ctx{};
+  auto gpu = stream_ctx.get_scheduler();
+  auto snd = ex::transfer_when_all(gpu, ex::just(3), ex::just(0.1415));
+  static_assert(ex::sender<decltype(snd)>);
+  (void)snd;
+}
+
+TEST_CASE("transfer_when_all with environment returns a sender",
+    "[cuda][stream][adaptors][transfer_when_all]") {
+  nvexec::stream_context stream_ctx{};
+  auto gpu = stream_ctx.get_scheduler();
+  auto snd = ex::transfer_when_all(gpu, ex::just(3), ex::just(0.1415));
+  static_assert(ex::sender<decltype(snd), empty_env>);
+  (void)snd;
+}
+
+// Hangs in `sync_wait`
+// TEST_CASE("transfer_when_all with no senders", "[cuda][stream][adaptors][transfer_when_all]") {
+//   nvexec::stream_context stream_ctx{};
+//   auto gpu = stream_ctx.get_scheduler();
+//   auto snd = ex::transfer_when_all(gpu);
+//   wait_for_value(std::move(snd));
+// }
```

Comment:
@trxcllnt is this bug filed. @senior-zero are you aware of this?

Reply:
@ericniebler Not yet, I just discovered it last night

Reply:
This is fixed.